flink-runtime/src/main/java/org/apache/flink/runtime/minicluster/MiniCluster.java

```diff
@@ -279,6 +282,10 @@ public void start() throws Exception {
                     miniClusterConfiguration.getRpcServiceSharing() == RpcServiceSharing.SHARED;
 
             try {
+                final PluginManager pluginManager =
+                        PluginUtils.createPluginManagerFromRootFolder(configuration);
```

Comment:
This pretty much doesn't do anything so I'd suggest to use `FileSystem#initialize(Configuration).`

Reply:
The `FileSystem#initialize(Configuration)` method is marked as deprecated I did not want to add a usage again.

Reply:
then you can also pass null for the PluginManager.

Reply:
Why is `null` sufficient? I think the `pluginManager` is needed such that files in the plugin folder are loaded at all. That's the only way to read/write to S3 in local setup. I'm probably missing something here.

Reply:
Filesystems can still be loaded through the FileSystemFactory service if they are on the classpath.
Plugins in local environments are a bit eh because there's a) no sane way to control the plugin directory (no, setting some internal env variable is not sane) and b) you must not add a `dependency` on said FS otherwise you lose all the benefits.

Reply:
Yes, but users are trying out their Flink job locally and use a `plugin` folder in their working directories for s3 (that works by default).

Reply:
Just from the last month: https://lists.apache.org/thread/kojvkx9ops9fmdkkcwsmpzkwrx94wcym https://lists.apache.org/thread/2dnlk7lc489ts20338cl3ll8gvfx6f21